### ISSUE TITLE
iOS: Right Menu doesn't show items

### ISSUE THREAD

**Issue by allistoncarlos** [XAM-Consulting/SlideOverKit#103]
I'm having an issue where the right menu just shows the overlay view in iOS. In Android, the menu shows correctly, as it follows. What am I doing wrong?

I've already implemented the PageImpInterfaceRendereiOS and I've put the SlideOverKit.iOS.SlideOverKit.Init(); in the FinishedLaunching.

I'm using Xamarin Forms 4.0.0.482894 and SlideOverKit 2.1.6.2...

Here's my MenuPage

`<t:SlideMenuView 
    xmlns="http://xamarin.com/schemas/2014/forms" 
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" 
    xmlns:t="clr-namespace:SlideOverKit" 
    xmlns:behaviors="clr-namespace:Project.Behaviors;assembly=Project"
    xmlns:templateSelector="clr-namespace:Project.DataSelectorTemplates;assembly=Project"
    xmlns:circleImage="clr-namespace:ImageCircle.Forms.Plugin.Abstractions;assembly=ImageCircle.Forms.Plugin"
    xmlns:viewModelBase="clr-namespace:Project.ViewModels.Base;assembly=Project"
    viewModelBase:ViewModelLocator.AutoWireViewModel="true"
    x:Class="Project.Views.MenuPage">
    <ContentPage.Resources>
        <ResourceDictionary>
            <Style x:Key="MenuListStyle" TargetType="{x:Type ListView}">
                <Setter Property="VerticalOptions" Value="Center" />
                <Setter Property="Margin" Value="0" />
            </Style>
            
            <DataTemplate x:Key="hasSubtitleTemplate">
                <ViewCell>
                    <StackLayout HeightRequest="65" Margin="0, 5, 0, 5" Orientation="Horizontal" HorizontalOptions="Start" VerticalOptions="Start">
                        <circleImage:CircleImage Source="Profile" WidthRequest="50" Aspect="AspectFill" HorizontalOptions="Start" />
                        <StackLayout VerticalOptions="CenterAndExpand" Orientation="Vertical">
                            <Label
                                Margin="0, 0, 5, 0"
                                Text="{Binding Title}"
                                TextColor="White"
                                HorizontalOptions="StartAndExpand"
                                FontAttributes="Bold"
                                FontFamily="Montserrat-Regular.ttf#Montserrat Regular"
                            />
                            <Label
                                Margin="0, 0, 5, 0"
                                Text="{Binding Subtitle}"
                                TextColor="White"
                                HorizontalOptions="StartAndExpand"
                                FontFamily="Montserrat-Regular.ttf#Montserrat Regular"
                            />
                        </StackLayout>
                    </StackLayout>
                </ViewCell>
            </DataTemplate>
            <DataTemplate x:Key="otherTemplate">
                <ViewCell>
                    <StackLayout HeightRequest="35" VerticalOptions="CenterAndExpand" Orientation="Vertical">
                        <Label
                            Margin="5, 0, 0, 0"
                            Text="{Binding Title}"
                            TextColor="White"
                            HorizontalOptions="StartAndExpand"
                            VerticalOptions="CenterAndExpand"
                            FontAttributes="Bold"
                            FontFamily="Montserrat-Regular.ttf#Montserrat Regular"
                        />
                    </StackLayout>
                </ViewCell>
            </DataTemplate>
            
            <templateSelector:MenuDataTemplateSelector x:Key="menuDataTemplateSelector"
                HasSubtitleTemplate="{StaticResource hasSubtitleTemplate}"
                OtherTemplate="{StaticResource otherTemplate}" />
        </ResourceDictionary>
    </ContentPage.Resources>
    
    <Grid Margin="5, 10, 5, 5">
        <ListView 
                x:Name="MenuList"
                SeparatorVisibility="None"
                CachingStrategy="RecycleElement"
                Style="{StaticResource MenuListStyle}"
                HasUnevenRows="True"
                ItemTemplate="{StaticResource menuDataTemplateSelector}">
        </ListView>
    </Grid>
</t:SlideMenuView>`

And the CS code behind file:
`using System.Collections.Generic;
using AprovaBrasil.ViewModels.Base;
using SlideOverKit;
using Xamarin.Forms;

namespace AprovaBrasil.Views
{
    public partial class MenuPage : SlideMenuView
    {
        public MenuPage()
        {
            InitializeComponent();

            // You must set IsFullScreen in this case, 
            // otherwise you need to set HeightRequest, 
            // just like the QuickInnerMenu sample
            this.IsFullScreen = true;
            // You must set WidthRequest in this case
            this.WidthRequest = 200;
            this.MenuOrientations = MenuOrientation.RightToLeft;

            // You must set BackgroundColor, 
            // and you cannot put another layout with background color cover the whole View
            // otherwise, it cannot be dragged on Android
            this.BackgroundColor = Color.FromHex("#414141");

            // This is shadow view color, you can set a transparent color
            this.BackgroundViewColor = Color.FromHex("#99000000");

            this.MenuList.ItemsSource = new List<MenuItem>
            {
                new MenuItem { Title = "Olá, Tiago", Subtitle = "Editar Perfil", MenuAction = MenuAction.Profile },
                new MenuItem { Title = "Ajuda", MenuAction = MenuAction.Help },
                new MenuItem { Title = "Sair", MenuAction = MenuAction.Logout }
            };

            this.MenuList.ItemTapped += (sender, e) =>
            {
                MessagingCenter.Send(this, MessageKeys.MenuItemTapped, e.Item as MenuItem);
            };
        }
    }
}`

![Simulator Screen Shot - iPhone Xs Max - 2019-06-07 at 11 05 21](https://user-images.githubusercontent.com/1048484/59113597-b4166680-891b-11e9-8642-ca8e882eb379.png)

![Captura de Tela 2019-06-07 às 11 22 08](https://user-images.githubusercontent.com/1048484/59113606-b8428400-891b-11e9-8cd0-3651d3f30713.png)

**Comment by samukasantos** [comment]
Hello @allistoncarlos  using in the information you passed I tried to reproduce your example by following your Layout and DataTemplateSelector. I preferred to create one with a slightly different name from MenuItem since this class exists in the core of Xamarin.Forms. My rendering on Android was as follows:

<img width="426" alt="Screen Shot 2019-06-08 at 5 50 15 pm" src="https://user-images.githubusercontent.com/5481198/59144207-90d0e180-8a17-11e9-93ef-b125748c5550.png">

And without any other style in iOS rendered this way:

<img width="394" alt="Screen Shot 2019-06-08 at 5 51 00 pm" src="https://user-images.githubusercontent.com/5481198/59144214-bc53cc00-8a17-11e9-8d34-65db1480224f.png">

After a minor change in the ViewCell style this way:

<img width="345" alt="Screen Shot 2019-06-08 at 6 06 06 pm" src="https://user-images.githubusercontent.com/5481198/59144246-40a64f00-8a18-11e9-95d6-42b90c8de8dd.png">

The result was this:

<img width="388" alt="Screen Shot 2019-06-08 at 5 54 29 pm" src="https://user-images.githubusercontent.com/5481198/59144250-5ca9f080-8a18-11e9-97b3-d3cb4e23d28a.png">

Remember that I used the default font of each platform.

**Comment by allistoncarlos** [comment]
Hi @samukasantos, thanks for your time and help! I've created another project from the scratch, and it worked on it... I though it was because the TabbedPage, but it wasn't.

But I've managed to make it work! I was trying to call the ShowMenu() method using my ViewModel through the MessagingCenter class. While in Android it worked without problems, in iOS I had to call it directly in the View. With some adjustments I could make the project look like the expected in both platforms.

Once again, thanks for all the help!
